operations/rack_resiliency/rack_resiliency.md

```diff
@@ -0,0 +1,484 @@
+
+#### 4 Auto deploy of RRS (Rack Resiliency Service) helm chart
+
+RRS helm chart consisting RRS pod with RRS API (CLI backend) and RMS (Resiliency Monitoring Service) containers will be deployed post `management-nodes-rollout` IUF step.
+
+Post `management-nodes-rollout`, as part of post hook, it will check if the Rack Resiliency is enabled and zones (`k8s`/ `Ceph`) are created. If present, it will then deploy the RRS chart otherwise skip to deploy.
+
+## Verification of RR
+
+### 1 Verify placement discovery of Master, Worker and Storage Management nodes
+
+Map the nodes (master, worker and storage) to the racks where they belong to by analysing the output of `/tmp/rack_info.txt` on any of the node.
+
+**Example output:**
+
+```text
+   {
+     "x3000": [
+       "ncn-m001",
+       "ncn-w001",
+       "ncn-s001",
+       "ncn-w004"
+     ],
+     "x3001": [
+       "ncn-m002",
+       "ncn-w002",
+       "ncn-s003",
+       "ncn-w005"
+     ],
+     "x3002": [
+       "ncn-m003",
+       "ncn-w003",
+       "ncn-w006",
+       "ncn-s002"
+     ]
+}
+```
+
+The above example clearly depicts that there are three racks in the CSM system - `x3000`, `x3001` and `x3002` and the corresponding master, worker and storage nodes were mapped to the rack ID's.
+
+### 2 `k8s` Zoning for Master and Worker Nodes
+
+Check the zones created by executing the following command:
+
+```bash
+kubectl get no -L topology.kubernetes.io/zone
+```
+
+Here the admin/ user has defined the `k8s` zone label prefix as `my-rack` and 3 racks (`x3000`, `x3001` and `x3002`) are present.
+
+**Example output:**
+
+```text
+NAME       STATUS   ROLES           AGE   VERSION    ZONE
+ncn-m001   Ready    control-plane   91d   v1.24.17   my-rack-x3000
+ncn-m002   Ready    control-plane   92d   v1.24.17   my-rack-x3001
+ncn-m003   Ready    control-plane   92d   v1.24.17   my-rack-x3002
+ncn-w001   Ready    <none>          92d   v1.24.17   my-rack-x3000
+ncn-w002   Ready    <none>          92d   v1.24.17   my-rack-x3001
+ncn-w003   Ready    <none>          92d   v1.24.17   my-rack-x3002
+ncn-w004   Ready    <none>          92d   v1.24.17   my-rack-x3000
+ncn-w005   Ready    <none>          92d   v1.24.17   my-rack-x3001
+ncn-w006   Ready    <none>          91d   v1.24.17   my-rack-x3002
+```
+
+Check the config for a specific zone:
+
+```bash
+kubectl get nodes -l topology.kubernetes.io/zone=**my-rack-x3002**
+```
+
+**Example output:**
+
+```text
+NAME       STATUS   ROLES           AGE   VERSION
+ncn-m003   Ready    <none>          42h   v1.24.17
+ncn-w003   Ready    <none>          42h   v1.24.17
+ncn-w006   Ready    <none>          42h   v1.24.17
+```
+
+We can also check the `k8s` labels applied for Master and Worker nodes: 
+
+```bash

```

Comment:
Either change this from `bash` to `text`, or split this into two boxes, the first being the command without the prompt (`bash`) and the second as `text`.  (Making this all `bash` causes the `<none>` to have the `<>` in red)